### PR TITLE
[#5455] feat(web): disable to update the reserved fields of table properties

### LINKED ISSUE #5455: [Subtask] Disable to update the reserved fields of table properties
### Describe the subtask

You can pass table properties to Gravitino when creating an table.

The Gravitino server doesn't allow passing the following reserved fields: https://gravitino.apache.org/docs/0.6.1-incubating/lakehouse-iceberg-catalog#table-properties

should disable to edit.

### Parent issue

https://github.com/apache/gravitino/issues/5449

### PR DESCRIPTION
**Pull request by orenccl**
### What changes were proposed in this pull request?

Disable the creation or update of reserved fields in table properties.

### Why are the changes needed?

Fix #5455.

### Does this PR introduce any user-facing changes?

Users will no longer be able to create or update reserved fields in table properties.

### How was this patch tested?

Tested by attempting to create or update reserved fields in table properties to ensure it is not allowed.

### CODE REVIEW

File: web/web/src/lib/utils/initial.js [hunk 1]
```diff
@@ -570,10 +566,64 @@ const relationalColumnTypeMap = {
...
+      'primary-key',
+      'partition'
+    ],
+    allowDelete: true,
+    allowAdd: true
+  }
+}
+
+export const getRelationalTablePropInfo = catalog => {
+  if (Object.keys(relationalTablePropInfoMap).includes(catalog)) {
```
Comment (tengqm) [inline web/web/src/lib/utils/initial.js:620]: How about
```suggestion
  if (Object.hasOwn(relationalTablePropInfoMap, catalog)) {
```

**Review by tengqm — COMMENTED** [review]
LGTM

File: web/web/src/lib/utils/initial.js (hunk 1 above)
Comment (orenccl) [reply web/web/src/lib/utils/initial.js]: This is a better approach and has already been updated.

File: web/web/src/lib/utils/initial.js [hunk 2]
```diff
@@ -570,10 +566,64 @@ const relationalColumnTypeMap = {
...
+  if (Object.hasOwn(relationalColumnTypeMap, catalog)) {
     return relationalColumnTypeMap[catalog]
   }
 
   return []
 }
+
+const relationalTablePropInfoMap = {
+  'lakehouse-iceberg': {
+    reserved: ['location'],
```
Comment (LauraXia123) [inline web/web/src/lib/utils/initial.js:579]: Aren't there some properties missing about iceberg: https://gravitino.apache.org/docs/0.7.0-incubating/lakehouse-iceberg-catalog#table-properties

**Comment by LauraXia123** [comment]
image
Would you mind optimizing the length of the input by the way?
And disabled the immutable properties
image

File: web/web/src/lib/utils/initial.js (hunk 2 above)
Comment (orenccl) [reply web/web/src/lib/utils/initial.js:579]: Sorry, I missed that. Will update it.

Some properties, like `serde-name`, are not in the documentation, but are they also immutable?

**Comment by LauraXia123** [comment]
> Hi @LauraXia123,
> 
> I’ve updated the properties field width and the reserved/immutable column settings.
> 
> However, I noticed that some properties appear to be immutable but are not mentioned in the documentation:
> 
> * **Hive**: `[serde-name]`
> * **Iceberg**: `[provider, format, format-version]`
> 
> I’ve marked them as immutable in the code, but I’m not certain if this is the correct approach. Could you please confirm?

Yes, these properties are immutable.

**Comment by orenccl** [comment]
Okay, then it's ready for review.

**Comment by LauraXia123** [comment]
Thank you for your contributions. LGTM

**Review by LauraXia123 — APPROVED** [review]

**Review by jerryshao — APPROVED** [review]

**Pull request merged**